web/tests/copy_and_paste.test.js

```diff
@@ -108,4 +122,12 @@ run_test("paste_handler", () => {
     insert_syntax_and_focus_called = false;
     copy_and_paste.paste_handler(event);
     assert.ok(!insert_syntax_and_focus_called);
+
+    // Pasting code block
+    input =
+        '<meta http-equiv="content-type" content="text/html; charset=utf-8"><div style="color: #ffffff;background-color: #002451;font-family: Consolas, &quot;Courier New&quot;, monospace;font-weight: normal;font-size: 14px;line-height: 19px;white-space: pre;"><div><span style="color: #ebbbff;">const</span><span style="color: #ffffff;"> </span><span style="color: #ff9da4;">compose_ui</span><span style="color: #ffffff;"> </span><span style="color: #99ffff;">=</span><span style="color: #ffffff;"> </span><span style="color: #bbdaff;">mock_esm</span><span style="color: #ffffff;">(</span><span style="color: #d1f1a9;">"../src/compose_ui"</span><span style="color: #ffffff;">);</span></div><div><span style="color: #bbdaff;">set_global</span><span style="color: #ffffff;">(</span><span style="color: #d1f1a9;">"document"</span><span style="color: #ffffff;">, </span><span style="color: #ff9da4;">document</span><span style="color: #ffffff;">);</span></div></div>';
+    assert.equal(
+        copy_and_paste.paste_handler_converter(input),
+        'const compose_ui = mock_esm("../src/compose_ui");\n\nset_global("document", document);',
```

Comment:
Hmm, is this a code block, or just the internals of the code block? I'd expect to possibly see a fence in the output if a full code block.

Reply:
This is when code is copied from an external source like VS Code or Gmail. Copying from here doesn't include the `code` element -- in fact, Gmail just has `pre` with a `div` inside it; no `code` element at all (which is exactly what the input here is like)